src/SDK/Logs/Processor/BatchLogsProcessor.php

```diff
@@ -0,0 +1,275 @@
+        }
+
+        $this->exporter = $exporter;
+        $this->clock = $clock;
+        $this->maxQueueSize = $maxQueueSize;
+        $this->scheduledDelayNanos = $scheduledDelayMillis * 1_000_000;
+        $this->maxExportBatchSize = $maxExportBatchSize;
+        $this->autoFlush = $autoFlush;
+
+        $this->exportContext = Context::getCurrent();
+        $this->queue = new SplQueue();
+        $this->flush = new SplQueue();
+
+        if ($meterProvider === null) {
+            return;
+        }
+
+        $meter = $meterProvider->getMeter('io.opentelemetry.sdk');
+        $meter
+            ->createObservableUpDownCounter(
+                'otel.logs.log_processor.logs',
+                '{logs}',
+                'The number of log records received by the processor',
+            )
+            ->observe(function (ObserverInterface $observer): void {
+                $queued = $this->queue->count() * $this->maxExportBatchSize + count($this->batch);
+                $pending = $this->queueSize - $queued;
+                $processed = $this->processed;
+                $dropped = $this->dropped;
+
+                $observer->observe($queued, self::ATTRIBUTES_QUEUED);
+                $observer->observe($pending, self::ATTRIBUTES_PENDING);
+                $observer->observe($processed, self::ATTRIBUTES_PROCESSED);
+                $observer->observe($dropped, self::ATTRIBUTES_DROPPED);
+            }, true);
+        $meter
+            ->createObservableUpDownCounter(
+                'otel.logs.log_processor.queue.limit',
+                '{logs}',
+                'The queue size limit',
+            )
+            ->observe(function (ObserverInterface $observer): void {
+                $observer->observe($this->maxQueueSize, self::ATTRIBUTES_PROCESSOR);
+            }, true);
+        $meter
+            ->createObservableUpDownCounter(
+                'otel.logs.log_processor.queue.usage',
+                '{logs}',
+                'The current queue usage',
+            )
+            ->observe(function (ObserverInterface $observer): void {
+                $queued = $this->queue->count() * $this->maxExportBatchSize + count($this->batch);
+                $pending = $this->queueSize - $queued;
+                $free = $this->maxQueueSize - $this->queueSize;
+
+                $observer->observe($queued, self::ATTRIBUTES_QUEUED);
+                $observer->observe($pending, self::ATTRIBUTES_PENDING);
+                $observer->observe($free, self::ATTRIBUTES_FREE);
+            }, true);
+    }
+
+    public function onEmit(ReadWriteLogRecord $record, ?ContextInterface $context = null): void
+    {
+        if ($this->closed) {
+            return;
+        }
+
+        if ($this->queueSize === $this->maxQueueSize) {
+            $this->dropped++;
+
+            return;
+        }
+
+        $record->setContext($context);
```

Comment:
IMO this should be done by the `Logger`, not within the processors.

Reply:
This is the way that I interpret the spec says to do it: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#onemit - the Logger passes a context on to the processor.

Reply:
I see what you mean. Java accepts but ignores `context`, having already added `spanContext` to the record. I guess that's still spec-compliant. Will make that change.